components/brave_rewards/resources/ui/components/checkout/addFundsPanel/index.tsx

```diff
@@ -0,0 +1,132 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import * as React from 'react'
+import { Button } from 'brave-ui/components'
+import { BatColorIcon } from 'brave-ui/components/icons'
+
+import { FormSection } from '../formSection'
+import { CreditCardForm, CreditCardFormHandle, CreditCardDetails } from '../creditCardForm'
+import { TermsOfSale } from '../termsOfSale'
+import { GoBackLink } from '../goBackLink'
+
+import {
+  Subtitle,
+  CurrentBalance,
+  ExchangeRateDisplay,
+  AmountOptionContainer,
+  AmountOptionExchange,
+  ChargeSummary,
+  PurchaseButtonRow
+} from './style'
+
+interface AddFundsAmountOption {
+  amount: number
+  amountConverted: string
+  transactionFeeRate: string
+  transactionFee: string
+  totalCharge: string
+}
+
+interface AddFundsPanelProps {
+  amountNeeded: number
+  walletBalance: number
+  walletBalanceConverted: string
+  unitValueConverted: string
+  amountOptions: AddFundsAmountOption[]
+  onCancel: () => void
+  onPayWithCreditCard: (cardDetails: CreditCardDetails) => void
+}
+
+export function AddFundsPanel (props: AddFundsPanelProps) {
+  if (props.amountOptions.length === 0) {
+    throw new Error('props.amountOptions cannot be an empty array')
+  }
+
+  const [selectedAmount, setSelectedAmount] = React.useState<number>(0)
+
+  let selectedOption = props.amountOptions[0]
+  for (const option of props.amountOptions) {
+    if (option.amount < props.amountNeeded) {
+      throw new Error('An amount in props.amountOptions is less than props.amountNeeded')
+    }
+    if (option.amount === selectedAmount) {
+      selectedOption = option
+      break
+    }
+  }
+
+  const creditCardFormRef = React.useRef<CreditCardFormHandle>(null)
+
+  const onPurchaseClick = () => {
+    const formHandle = creditCardFormRef.current
+    if (formHandle) {
+      const errors = formHandle.validate()
+      if (errors.length === 0) {
+        props.onPayWithCreditCard(formHandle.details)
+      } else {
+        errors[0].element.focus()
+      }
+    }
+  }
+
+  return (
+    <>
+      <h1>Add Funds</h1>
+      <Subtitle>
+        Add BAT to your wallet using your credit card
+      </Subtitle>
+      <CurrentBalance>
+        <div>
+          Current balance
+          <span className='balance-bat'>{props.walletBalance.toFixed(1)} BAT</span>
```

Comment:
wallet balance should already be formatted  when it comes in

Reply:
Done. I've kept the `walletBalance` as a number in `PaymentMethodPanel` though, as it compares that value with the total order amount.